app/controllers/api/v1/discussions_users_controller.rb

```diff
@@ -0,0 +1,36 @@
+class API::V1::DiscussionsUsersController < API::V1::ApplicationController
+
+  before_action :load_discussion, only: [:create, :destroy]
+  before_action :load_user, only: [:create, :destroy]
+  before_action :check_prior_rsvp, only: [:create, :destroy]
+
+  def create
+    if @user.discussions_users.create(discussion: @discussion)
+      render json: { success: "You have successfully RSVP'd to this discussion", status: 200 }
+    else
+      render json: { failure: "Couldn't RSVP to the discussion", status: 412 }
+    end
+  end
+
+  def destroy
+    if @user.discussions_users.find_by(discussion: @discussion).destroy
+      render json: { success: 'You opted out of this discussion', status: 200 }
+    else
+      render json: { failure: "Couldn't opt out of the discussion", status: 412 }
+    end
+  end
+
+  private
+
+    def check_prior_rsvp
+      if DiscussionsUser.find_by(user: @user, discussion: @discussion)
+        render json: { notice: 'You are already #{ "not" if action_name == "destroy" } attending this discussion' }
```

Comment:
what if this is before action for some other action too.. and the message I need is same as in case of destroy action. Messages should depend on conditions of state of DiscussionUser,, not action_name.